05_debug: Kernel debugging homework

Rampopula opened the pull request:
GL 05_debug kernel loadable module task and solution

ekovalyov committed:
Homework: Kernel debugging

Commit:
05_debug: Add setup environment script, add project skeleton

Added setup.sh script to set environment for module building and installing
Added project skeleton

Signed-off-by: Eduard Chaika <eduard.chaika@globallogic.com>

alexposukhov commented:
Please check you C and h files with checkpatch to make sure it matches linux code style:
checkpatch.pl script is located in linux-stable/scripts folder run it with -no-tree and -f options: for example:
#linux-stable$ ./scripts/checkpatch.pl --no-tree -f ~/gl-kernel-training-2021/05_debug/debug_comm.c

Please divide commit description on two lines to be sure that every line length is no longer 80 characters.

Commit:
05_debug: Add statistics collect, string store and convert

Implemented software module for collecting statistics, storing and converting
input strings

Signed-off-by: Eduard Chaika <eduard.chaika@globallogic.com>

Commit:
05_debug: Implement procfs converter

Implemented procfs character-to-uppercase coverter.

Signed-off-by: Eduard Chaika <eduard.chaika@globallogic.com>

Commit:
05_debug: Implement sysfs converter

Implemented sysfs character-to-lowercase coverter.

Signed-off-by: Eduard Chaika <eduard.chaika@globallogic.com>
